src/object/object_primitive.c

```diff
@@ -15700,3 +16100,115 @@ mr_cmpval_enumeration (DB_VALUE * value1, DB_VALUE * value2, int do_coercion, in
+ * data(out)					  : The result of the decompression. !!! Needs to be alloc'ed !!!
+ * compressed_size(in)				  : The compressed data size.
+ * decompressed_size(in)			  : The uncompressed data size.
+ */
+int
+mr_get_compressed_data_from_buffer (OR_BUF * buf, char *data, int compressed_size, int decompressed_size)
+{
+  int rc = NO_ERROR;
+
+  /* Check if the string needs decompression */
+  if (compressed_size > 0)
+    {
+      lzo_uint decompression_size = 0;
+      /* Handle decompression */
+
+      /* decompressing the string */
+      rc = lzo1x_decompress ((lzo_bytep) buf->ptr, (lzo_uint) compressed_size, data, &decompression_size, NULL);
+      if (rc != LZO_E_OK)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_LZO_DECOMPRESS_FAIL, 1, 0);
+	  return ER_IO_LZO_DECOMPRESS_FAIL;
+	}
+      if (decompression_size != decompressed_size)
+	{
+	  /* Decompression failed. It shouldn't. */
+	  assert (false);
+	}
+      data[decompressed_size] = '\0';
+    }
+  else
+    {
+      /* String is not compressed and buf->ptr is pointing towards an array of char's of length equal to
+       * decompressed_size */
+
+      rc = or_get_data (buf, data, decompressed_size);
+      data[decompressed_size] = '\0';
+    }
+
+  return rc;
+}
+
+/* 
+ * mr_get_compression_length()		  - Simulate a compression to find its length to be stored on the disk.
+ * 
+ * return()				  : The length of the compression, based on the new encoding of varchar.
+ *					    If the compression fails, then it returns charlen.
+ * string(in)				  : The string to be compressed.
+ * charlen(in)				  : The length of the string to be compressed.
+ */
+int
+mr_get_compression_length (char *string, int charlen)
+{
+  lzo_voidp wrkmem;
+  char *compressed_string = NULL;
+  lzo_uint compressed_length = 0;
+  int rc = NO_ERROR;
+
+  wrkmem = (lzo_voidp) malloc (LZO1X_1_MEM_COMPRESS);
+  if (wrkmem == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, (size_t) LZO1X_1_MEM_COMPRESS);
+      goto cleanup;
+    }
+
+  /* Alloc memory for the compressed string */
+  /* Worse case LZO compression size from their FAQ */
+  compressed_string = (char *) malloc ((charlen + (charlen / 16) + 64 + 3) * sizeof (char));
+  if (compressed_string == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY,
+	      1, (size_t) (charlen + (charlen / 16) + 64 + 3));
+      goto cleanup;
+    }
+
+  /* Compress the string */
+  rc = lzo1x_1_compress (string, charlen, compressed_string, &compressed_length, wrkmem);
+  if (rc != LZO_E_OK)
+    {
+      /* We should not be having any kind of errors here. Because if this compression fails, there is not warranty
+       * that the compression from putting data into buffer will fail as well. This needs to be checked but for now
+       * we leave it as it is.
+       */
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_LZO_COMPRESS_FAIL, 1, 0);
+      goto cleanup;
```

Comment:
@lucianlazar2211 

Please notice the message of `ER_IO_LZO_COMPRESS_FAIL` has 4 arguments:

```
896 Compression failed. Zip Method: %1$d (%2$s), Zip Level: %3$d (%4$s)
```